src/Microsoft.AspNet.Mvc.TagHelpers/Internal/FileProviderGlobbingDirectory.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.AspNet.FileProviders;
+using Microsoft.Framework.FileSystemGlobbing.Abstractions;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers.Internal
+{
+    public class FileProviderGlobbingDirectory : DirectoryInfoBase
+    {
+        private readonly IFileProvider _fileProvider;
+        private readonly IFileInfo _fileInfo;
+        private readonly FileProviderGlobbingDirectory _parent;
+
+        public FileProviderGlobbingDirectory(
+            IFileProvider fileProvider,
+            IFileInfo fileInfo,
+            FileProviderGlobbingDirectory parent)
+        {
+            _fileProvider = fileProvider;
+            _fileInfo = fileInfo;
+            _parent = parent;
+            if (_parent != null && !string.IsNullOrEmpty(_parent.RelativePath) && _fileInfo != null)
+            {
+                RelativePath = _parent.RelativePath + Path.DirectorySeparatorChar + _fileInfo.Name;
```

Comment:
Is using `Path.DirectorySeparatorChar` the right thing here? Or is the Use of fileprovider somehow normalizes the paths to use a common separator.